nav2_collision_monitor/src/pointcloud.cpp

```diff
@@ -106,8 +106,27 @@ bool PointCloud::getData(
   sensor_msgs::PointCloud2ConstIterator<float> iter_y(*data_, "y");
   sensor_msgs::PointCloud2ConstIterator<float> iter_z(*data_, "z");
 
+  bool height_present{false};
+  for(const auto & field : data_->fields) {
+    if(field.name == "height") {
+      height_present = true;
+    }
+  }
+
+  // Reference height field
+  std::string height_field{""};
+  if(use_global_height_ && height_present) {
+    height_field = "height";
+  } else if(use_global_height_) {
+    throw std::runtime_error{"'use_global_height' parameter true but height field not in cloud"};
```

Comment:
I do not believe that there is a catch anywhere, so this would actually bring down the server or put lifecycle into the error handling state. This should log and error and return `false`. That should cause the appropriate fallback situation https://github.com/ros-navigation/navigation2/blob/11f04c708e2c191a4eb3947ee38e9244844b87f8/nav2_collision_monitor/src/collision_monitor_node.cpp#L446-L455